lib/client.js

```diff
@@ -55,13 +55,22 @@ var SparqlClient = module.exports = function SparqlClient(endpoint, options) {
                     throw error;
                 }
 
-                responseBody = JSON.parse(response.body);
+                responseBody = parseJSON(response.body);
 
                 if (queryOptions) {
                     formatter.format(responseBody, queryOptions);
                 }
                 return promise.resolve(responseBody);
 
+                function parseJSON(body) {
+                    // Catch invalid JSON
+                    try {
+                        return JSON.parse(response.body);
```

Comment:
Sorry about that, I'll try not to rush these PR's to much anymore and review them more carefully next time.